pkg/client/client.go

```diff
@@ -0,0 +1,39 @@
+package client
+
+import (
+	"context"
+	"log/slog"
+	"os"
+
+	camelv1 "github.com/apache/camel-k/pkg/apis/camel/v1"
+	camelclient "github.com/apache/camel-k/pkg/client"
+	"github.com/go-logr/logr/slogr"
+	controllerlog "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+var logger = slog.Default()
+
+type Client struct {
+	camelclient.Client
+}
+
+func NewClient() (*Client, error) {
+	controllerlog.SetLogger(slogr.NewLogr(logger.Handler()))
+
+	cl, err := camelclient.NewClient(false)
+	if err != nil {
+		logger.Error("failed to create k8s client for camel k", err)
+		os.Exit(1)
```

Comment:
may be better to return the error and handle the error in the caller instead of `os.Exit`

Reply:
https://github.com/sco1237896/sco-backend/pull/1/commits/8ba8844d03461c3c11cc8e084e6fae3bc42283ff